plugins/drivers/base/driver.go

```diff
@@ -0,0 +1,194 @@
+}
+
+type HealthState string
+
+var (
+	HealthStateUndetected = HealthState("undetected")
+	HealthStateUnhealthy  = HealthState("unhealthy")
+	HealthStateHealthy    = HealthState("healthy")
+)
+
+type Fingerprint struct {
+	Attributes        map[string]string
+	Health            HealthState
+	HealthDescription string
+}
+
+type FSIsolation string
+
+var (
+	FSIsolationNone   = FSIsolation("none")
+	FSIsolationChroot = FSIsolation("chroot")
+	FSIsolationImage  = FSIsolation("image")
+)
+
+type Capabilities struct {
+	// SendSignals marks the driver as being able to send signals
+	SendSignals bool
+
+	// Exec marks the driver as being able to execute arbitrary commands
+	// such as health checks. Used by the ScriptExecutor interface.
+	Exec bool
+
+	//FSIsolation indicates what kind of filesystem isolation the driver supports.
+	FSIsolation FSIsolation
+}
+
+type TaskConfig struct {
+	ID              string
+	Name            string
+	Env             map[string]string
+	Resources       Resources
+	Devices         []DeviceConfig
+	Mounts          []MountConfig
+	User            string
+	AllocDir        string
+	rawDriverConfig []byte
+}
+
+func (tc *TaskConfig) EnvList() []string {
+	l := make([]string, len(tc.Env))
+	for k, v := range tc.Env {
+		l = append(l, k+"="+v)
+	}
+	return l
+}
+
+func (tc *TaskConfig) TaskDir() *allocdir.TaskDir {
+	taskDir := filepath.Join(tc.AllocDir, tc.Name)
+	return &allocdir.TaskDir{
+		Dir:            taskDir,
+		SharedAllocDir: filepath.Join(tc.AllocDir, allocdir.SharedAllocName),
+		LogDir:         filepath.Join(tc.AllocDir, allocdir.SharedAllocName, allocdir.LogDirName),
+		SharedTaskDir:  filepath.Join(taskDir, allocdir.SharedAllocName),
+		LocalDir:       filepath.Join(taskDir, allocdir.TaskLocal),
+		SecretsDir:     filepath.Join(taskDir, allocdir.TaskSecrets),
+	}
+}
+
+func (tc *TaskConfig) DecodeDriverConfig(t interface{}) error {
+	return base.MsgPackDecode(tc.rawDriverConfig, t)
+}
+
+func (tc *TaskConfig) EncodeDriverConfig(t interface{}) error {
+	return base.MsgPackEncode(&tc.rawDriverConfig, t)
+}
+
+type Resources struct {
+	CPUPeriod        int64
+	CPUQuota         int64
+	CPUShares        int64
+	MemoryLimitBytes int64
+	OOMScoreAdj      int64
+	CpusetCPUs       string
+	CpusetMems       string
+}
+
+type DeviceConfig struct {
+	TaskPath    string
+	HostPath    string
+	Permissions string
+}
+
+type MountConfig struct {
+	TaskPath string
+	HostPath string
+	Readonly bool
+}
+
+const (
+	TaskStateUnknown TaskState = "unknown"
+	TaskStateRunning TaskState = "running"
+	TaskStateExited  TaskState = "exited"
+)
+
+type TaskState string
+
+type NetworkOverride struct {
+	PortMap       map[string]int32
+	Addr          string
+	AutoAdvertise bool
+}
+
+type ExitResult struct {
+	ExitCode  int
+	Signal    int
+	OOMKilled bool
+	Err       error
+}
+
+type TaskStatus struct {
+	ID               string
+	Name             string
+	State            TaskState
+	SizeOnDiskMB     int64
```

Comment:
From its placement in TaskStatus instead of TaskStats I assume this is EphemeralDisk.Size? I think we should name it `EphemeralDiskSize` if so to avoid confusion as to whether this is the static allocation vs a runtime measurement.

Reply:
This is meant to be a runtime measurement of the current disk usage.

Reply:
Shouldn't it be in Stats with the other runtime measurements then? I don't think we've implemented this anywhere yet, so it can probably be removed entirely.

Reply:
The original plan was that today InspectTask is actually never called, but we'd like to have a cli (and match UI) that allows you to inspect a task and get all the delicious debug information you need from it. This was one value that I thought should be included here.